### ISSUE TITLE
Export multiple files from client generators

### ISSUE THREAD

**Issue by mxstbr** [dotansimha/graphql-code-generator#1815]
**Is your feature request related to a problem? Please describe.**

When generating client-side components (e.g. with the react-apollo or upcoming urql generators), all the query components and hooks are put into a single file.

That is bad news for code-splitting, as many of these components might only be used on a single page but now all of them will always be bundled on every page.

**Describe the solution you'd like**

It would be great if I could generate one file next to every query I have. For example:

```yaml
documents: "src/**/*.graphql"
generates:
  whatever.tsx:
    config:
      relativeFiles: true
    plugins:
      - "typescript-urql"
```

This would now create a component file next to every `.graphql` file in my project, and I could import them each individually.

**Comment by dotansimha** [comment]
Hi @mxstbr :) 

It's done in: https://github.com/dotansimha/graphql-code-generator/pull/1846 (still experimental and it will warn about it in the stable release, but I tested it with react-apollo, apollo-angular, url and stencil-apollo plugins and all seems to work, also with fragments imports)

**Comment by dotansimha** [comment]
Available in `1.2.0`.
